File: Integral_process_demo/server/src/main/java/com/yzq/Integral_process_server/server/ChangePwdServlet.java
package com.yzq.Integral_process_server.server;

import com.alibaba.druid.pool.DruidPooledConnection;
import com.yzq.Integral_process_server.server.utils.DruidUtils;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.sql.PreparedStatement;
import java.sql.SQLException;


/**
 * @author : yuzhiqiang (zhiqiang.yu.xeon@gmail.com)
 * @description: 更改密码 （更改数据库中的记录）
 * @date : 2021/4/4
 * @time : 17:09
 */
@WebServlet(name = "ChangePwdServlet", value = "/changePwd")
public class ChangePwdServlet extends HttpServlet {

    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        String account = request.getParameter("account");
        System.out.println("account = " + account);
        String pwd = request.getParameter("pwd");
        System.out.println("pwd = " + pwd);

        try {
            DruidPooledConnection connection = DruidUtils.druidDataSource.getConnection();
            String sql = "update account set pwd=? where user_name=?";
            PreparedStatement preparedStatement = connection.prepareStatement(sql);

            preparedStatement.setString(1, pwd);
            preparedStatement.setString(2, account);


            int i = preparedStatement.executeUpdate();

            System.out.println("i = " + i);

            if (i > 0) {
                response.getWriter().write("修改成功");
            } else {
                response.getWriter().write("修改失败");
            }

            connection.close();


        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }

    }
}

File: Integral_process_demo/kotlinServer/src/main/kotlin/com/yzq/kotlin_server/kotlinServer/servlet/ChangePwdServlet.kt
package com.yzq.kotlin_server.kotlinServer.servlet

import com.yzq.kotlin_server.kotlinServer.utils.DruidUtils
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse


/**
 * @description: 修改密码 对应数据库中的 改
 * @author : yuzhiqiang (zhiqiang.yu.xeon@gmail.com)
 * @date   : 2021/4/4
 * @time   : 20:49
 */
@WebServlet(name = "ChangePwdServlet", value = ["/changePwd"])
class ChangePwdServlet : HttpServlet() {

    override fun doPost(request: HttpServletRequest, response: HttpServletResponse) {

        val account = request.getParameter("account")
        println("account = ${account}")
        val pwd = request.getParameter("pwd")
        println("pwd = ${pwd}")


        val sql = "update account set pwd=?  where user_name=?"

        try {

            val connection = DruidUtils.druidDataSource.getConnection()
            val prepareStatement = connection.prepareStatement(sql)
            prepareStatement.setString(1, pwd)
            prepareStatement.setString(2, account)

            val executeUpdate = prepareStatement.executeUpdate()

            if (executeUpdate > 0) {
                response.writer.write("密码修改成功")
            } else {
                response.writer.write("密码修改失败")
            }

            connection.close()
        } catch (e: Exception) {
            e.printStackTrace()
            response.writer.write("密码修改失败")
        } finally {

        }


    }
}